apps/examples/webserver_test/webserver_test_main.c

```diff
@@ -0,0 +1,1308 @@
+				goto errout;
+			}
+
+			param->response->method = param->method;
+			param->response->url = param->url;
+
+			if (param->callback && param->response->entity_len != 0) {
+				param->callback(param->response);
+			}
+		}
+	}
+
+	if (param->callback) {
+		ws_test_http_client_response_release(param->response);
+	}
+	
+	if (param->tls) {
+		ws_test_wget_tls_release(client_tls);
+		ws_test_wget_tls_ssl_release(client_tls);
+	}
+	free(client_tls);
+	if (!param->tls) {
+		close(sockfd);
+	}
+	param->async_flag = WGET_OK;
+	FUNC_EX;
+	return WGET_OK;
+
+errout:
+	if (param->callback && param->response) {
+		ws_test_http_client_response_release(param->response);
+	}
+	if (param->tls) {
+		ws_test_wget_tls_ssl_release(client_tls);
+	}
+errout_before_tlsinit:
+	if (param->tls) {
+		ws_test_wget_tls_release(client_tls);
+	}
+
+	free(client_tls);
+	if (!param->tls && sockfd > 0) {
+		close(sockfd);
+	}
+	param->async_flag = WGET_ERR;
+	FUNC_EX;
+	return WGET_ERR;
+}
+#endif
+
+int ws_test_client_send_request(struct http_client_request_t *request)
+{
+	FUNC_EN;
+	if (request == NULL) {
+		PRNT("Error: Request is null");
+		return -1;
+	}
+	if (request->method < WGET_MODE_GET || request->method > WGET_MODE_DELETE) {
+		PRNT("Error: Incorrect method value!!");
+		return -1;
+	}
+	if (request->buffer == NULL) {
+		PRNT("Error: Buffer is NULL!!");
+		return -1;
+	}
+	if (request->buflen <= 0) {
+		PRNT("Error: Buffer length must be bigger than 0!!");
+		return -1;
+	}
+	if (request->encoding != CONTENT_LENGTH && request->encoding != CHUNKED_ENCODING) {
+		PRNT("Error: Incorrect encoding value!!");
+		return -1;
+	}
+
+	// sync call
+	int ret = ws_test_wget_base((void *)request);
+	FUNC_EX;
+	return ret;
+}
+
+static int ws_test_http_client_send_request(struct http_client_request_t *request,
+		 void *ssl_config, struct http_client_response_t *response, wget_callback_t cb)
+{
+	FUNC_EN;
+	int res = -1;
+#ifdef CONFIG_NET_SECURITY_TLS
+	struct mallinfo data;
+	struct http_client_ssl_config_t *ssl_conf = ssl_config;
+#endif
+
+	if (request == NULL) {
+		PRNT("Error: Request is null");
+		goto ret_error;
+	}
+
+	request->tls = false;
+	request->response = response;
+	request->callback = cb;
+
+	if (request->url == NULL) {
+		PRNT("Error: URL is NULL!!");
+		goto ret_error;
+	}
+	if (WEBCLIENT_CONF_MAX_ENTITY_SIZE < strlen(request->entity)) {
+		PRNT("WEBCLIENT_CONF_MAX_ENTITY_SIZE < strlen(request->entity)");
+		goto ret_error;
+	}
+#ifdef CONFIG_NET_SECURITY_TLS
+	if (ssl_conf) {
+		request->tls = true;
+		memcpy(&request->ssl_config, ssl_conf, sizeof(struct http_client_ssl_config_t));
+
+		data = mallinfo();
+		if (data.fordblks < WEBCLIENT_CONF_MIN_TLS_MEMORY) {
+			PRNT("Error: Not enough memory!!");
+			goto ret_error;
+		}
+	}
+#endif
+
+	request->buffer = (char *)zalloc(request->buflen);
+	if (request->buffer == NULL) {
+		PRNT("Error: Fail to allocate buffer");
+		return -1;
+	}
+
+	res = ws_test_client_send_request(request);
```

Comment:
ws_test_client_send_request can be removed as it is called from single place
required checks can be moved to starting of calling function
and ws_test_wget_base API can be also be called directly.